src/javascript/Switch.js

```diff
@@ -108,23 +114,43 @@ export class Switch {
             this.checkboxBlurred(e);
         }.bind(this), false);
 
-        this.element.addEventListener("click", function(e) {
+        this.element.addEventListener("click", function (e) {
+            if (this.disabled) {
+                e.preventDefault();
+                return
+            }
+
             this.checkboxToggled(e);
         }.bind(this), false);
+
+
+        // Bind changes to the disabled attribute
+        const observer = new MutationObserver((mutations) => {
+            mutations.forEach((mutation) => {
+                // Check the modified attributeName is "disabled"
+                if (mutation.attributeName === "disabled") {
```

Comment:
You are limiting this to attributes, but as a sanity check, you may want to add:

```javascript
if (mutation.type !== "attributes") {
  continue;
}
```

(which would obviously require refactoring the `foreach` to `for (const mutation of mutations)` or similar)